src/main/java/com/commercetools/sync/commons/models/WaitingToBeResolved.java

```diff
@@ -1,50 +1,28 @@
 package com.commercetools.sync.commons.models;
 
-import io.sphere.sdk.products.ProductDraft;
+
+import io.sphere.sdk.models.WithKey;
 
 import javax.annotation.Nonnull;
 import java.util.Objects;
 import java.util.Set;
 
-public final class WaitingToBeResolved {
-    private ProductDraft productDraft;
-    private Set<String> missingReferencedProductKeys;
-
-    /**
-     * Represents a productDraft that is waiting for some product references, which are on this productDraft as
-     * attributes, to be resolved.
-     *
-     * @param productDraft       product draft which has irresolvable references as attributes.
-     * @param missingReferencedProductKeys product keys of irresolvable references.
-     */
-    public WaitingToBeResolved(
-        @Nonnull final ProductDraft productDraft,
-        @Nonnull final Set<String> missingReferencedProductKeys) {
-        this.productDraft = productDraft;
-        this.missingReferencedProductKeys = missingReferencedProductKeys;
-    }
+public abstract class WaitingToBeResolved<T extends WithKey> {
+    private Set<String> missingReferencedKeys;
 
-    // Needed for the 'com.fasterxml.jackson' deserialization, for example, when fetching
-    // from CTP custom objects.
     public WaitingToBeResolved() {
     }
 
-    @Nonnull
-    public ProductDraft getProductDraft() {
-        return productDraft;
+    public Set<String> getMissingReferencedKeys() {
+        return missingReferencedKeys;
     }
 
-    @Nonnull
-    public Set<String> getMissingReferencedProductKeys() {
-        return missingReferencedProductKeys;
-    }
+    public abstract void setWaitingDraft(@Nonnull final T draft);
```

Comment:
this setter is not used.

Reply:
It is used to serialize this from a custom object